extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -182,6 +188,18 @@ class AmpViqeoPlayer extends AMP.BaseElement {
       } else {
         this.element.dispatchCustomEvent(VideoEvents.UNMUTED);
       }
+    } else if (action === 'end') {
```

Comment:
- Use `redispatch` from 'src/iframe-video' with a map rather than a large else-if sequence.
- It looks like `update` actions can be kept in a map instead rather than local props?

```js
const EVENTS = {
  'ready': VideoEvents.LOAD,
  'play': VideoEvents.PLAYING,
  // ...
  'startAdvert': VideoEvents.AD_START,
  'endAdvert': VideoEvents.AD_END,
  // ...
};

// ...

constructor(element) {
  // ...
  /** @private {Object<string, (number|Array)>} */
  this.meta_ = {};

  // ...
}

handleViqeoMessages_(event) {
  // ..
  if (redispatch(this.element, action, EVENTS)) {
    return;
  }
  if (startsWith(action, 'update')) {
    const key = action.replace(
      /^update([A-Z])(.*)$/,
      (_, c, rest) => c.toLowerCase() + rest
    )
    this.meta_[key] = eventData['value'];
  }
}

getCurrentTime() {
  return this.meta_['currentTime'] || 0;
}

getDuration() {
  return this.meta_['duration'] || 0;
}

getPlayedRanges() {
  return this.meta_['playedRanges'] || [];
}
```

Reply:
done